src/rules/a11y-explicit-heading.js

```diff
@@ -15,10 +15,12 @@ const isUsingAsProp = elem => {
 
 const isValidAsUsage = value => validHeadings.includes(value.toLowerCase())
 const isInvalid = elem => {
+  if (elem.attributes.length === 1 && elem.attributes[0].type === 'JSXSpreadAttribute') return;
```

Comment:
I noticed a bug when `<Heading>` was used with spread props (`{...args}`), or when passing a variable with`as` (`<Heading as={as}>`).

If the spread is the only attribute present on the component, the checks will skip it. Similarly, if the component is using a variable for the `as` prop, checks will also skip it.